trikot-kword/kword-remote-update/src/commonMain/kotlin/com/mirego/trikot/kword/remote/update/internal/RemoteTranslationsFetcher.kt

```diff
@@ -0,0 +1,100 @@
+package com.mirego.trikot.kword.remote.update.internal
+
+import com.mirego.trikot.kword.I18N
+import io.ktor.client.HttpClient
+import io.ktor.client.call.body
+import io.ktor.client.request.get
+import io.ktor.http.isSuccess
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.IO
+import kotlinx.coroutines.async
+import kotlinx.coroutines.awaitAll
+import kotlinx.coroutines.launch
+import kotlinx.serialization.json.Json
+
+internal class RemoteTranslationsFetcher(
+    private val baseTranslationsUrl: String?,
+    private val translationsVersion: String?,
+    private val internalCacheWrapper: InternalCacheWrapper
+) {
+    fun fetchRemoteTranslations(
+        i18N: I18N,
+        baseFileName: String,
+        baseMap: MutableMap<String, String>,
+        languageCodes: List<String>
+    ) {
+        baseTranslationsUrl?.let { baseTranslationsUrlVal ->
+            translationsVersion?.let { translationsVersionVal ->
+                if (baseTranslationsUrlVal.isEmpty() || translationsVersionVal.isEmpty()) {
+                    return
+                }
+
+                val sharedHttpClient = HttpClient()
+                CoroutineScope(Dispatchers.IO).launch {
```

Comment:
nitpick: the dispatcher could be a class argument to help for tests (is there any tests for this class?) . It can default to IO.

The dispatcher can then also be provided by updateTranslations to pass it here. Will allow the callers to define their own dispatcher.